src/core/Akka.Streams.Tests/IO/FileSinkSpec.cs

```diff
@@ -209,6 +248,29 @@ public void SynchronousFileSink_should_allow_overriding_the_dispatcher_using_Att
             }, _materializer);
         }
 
+        // Needed help converting this test case
+        [Fact]
+        public void SynchronousFileSink_should_write_single_line_to_a_file_from_lazy_sink()
+        {
+            this.AssertAllStagesStopped(() => {
+                TargetFile(f => {
+                    var lazySink = Sink.LazySink(
+                        (ByteString _) => Task.FromResult(FileIO.ToFile(f)),
+                        () => Task.FromResult(IOResult.Success(0)))
+                        .MapMaterializedValue(t => {
+                            t.Wait(TimeSpan.FromSeconds(3));
+                            return t.Result;
```

Comment:
I think it's more like `t.ContinueWIth(r=>r.Result.Result)` but not sure if it matters, if the test passes leave it as it is ;)